backend/post-service/src/main/java/com/speakapp/postservice/repositories/PostReactionRepository.java

```diff
@@ -12,5 +12,6 @@
 public interface PostReactionRepository extends JpaRepository<PostReaction, UUID> {
 
     List<PostReaction> findAllByPost(Post post);
+    PostReaction findByPostAndUserId(Post post, UUID userId);
```

Comment:
To powinno byc Optional<PostReaction>, ponieważ może dany user nie mieć żadnej reakcji na posta.

Reply:
W sensie patrząć na twój kod dalej jest git, ale jak ktoś inny będzie chciał z tego skorzystać to może zapomnieć o sprawdzaniu nulla